task/sast-coverity-buildless-check/0.1/sast-coverity-buildless-check.yaml

```diff
@@ -0,0 +1,193 @@
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: SEVERITY_THRESHOLD
+          value: $(params.SEVERITY_THRESHOLD)
+      script: |
+        #!/usr/bin/env bash
+        set -eo pipefail
+        . /usr/local/share/konflux-test/utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        echo 'Starting Coverity buildless scan'
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)
+        COVERITY_DIR=/tmp/cov-scan/cov
+        COVERITY_RESULTS_FILE=coverity-buildless-results.js
+        COV_LICENSE_PATH=/etc/secrets/cov-license
+        # Installing Coverity license
+        cp /etc/secrets/cov-license /opt/coverity/bin/license.dat
+        if test -z "$(find /opt -name 'license.dat')"; then
+          echo "No license file for Coverity was detected. Exiting..."
+          exit 1
+        fi
+        
+        # Installation of Red Hat certificates for cloning Red Hat internal repositories
+        curl -sS https://certs.corp.redhat.com/certs/2015-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2015-RH-IT-Root-CA.crt
+        curl -sS https://certs.corp.redhat.com/certs/2022-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2022-IT-Root-CA.pem
+        update-ca-trust
+        
+        # Create configuration file for coverity buildless
+        echo -e 'capture:\n   build-command-inference: false' > $SOURCE_CODE_DIR/coverity.yml
+        
+        # Captureless scan
+        env COV_HOST=konflux HOME=/tmp/cov-scan/home /opt/coverity/bin/coverity capture $COV_CAPTURE_ARGS --project-dir $SOURCE_CODE_DIR --dir $COVERITY_DIR
+        COV_CAPTURE_EXIT_CODE=$?
+        
+        if [[ "$COV_CAPTURE_EXIT_CODE" -eq 0 ]]; then
+            echo "Coverity capture scan finished succesfully"
+        else
+            echo "Coverity capture command failed. Exiting..."
+            exit 1
+        fi
+        
+        # Analysis phase
+        /opt/coverity/bin/cov-manage-emit --dir $COVERITY_DIR reset-host-name
+        /opt/coverity/bin/cov-analyze $COV_ANALYZE_ARGS --dir=$COVERITY_DIR --enable HARDCODED_CREDENTIALS --security --concurrency --spotbugs-max-mem=4096
+        COV_ANALYZE_EXIT_CODE=$?
+        
+        if [[ "$COV_ANALYZE_EXIT_CODE" -eq 0 ]]; then
+          echo "Coverity analyze scan finished successfully"
+        else
+          echo "Coverity analyze scan failed. Exiting..."
+          exit 1
+        fi
+        
+        /opt/coverity/bin/cov-format-errors --dir=$COVERITY_DIR --json-output-v10 $COVERITY_RESULTS_FILE
+        # We parse the results, embed context, remove duplicates and store them in SARIF format.
+        csgrep --mode=json --prepend-path-prefix=$SOURCE_CODE_DIR/ $COVERITY_RESULTS_FILE | csgrep --mode=json $SEVERITY_THRESHOLD --remove-duplicates --embed-context 3 | csgrep --mode=json --strip-path-prefix="$SOURCE_CODE_DIR"/source/ | csgrep --mode=json --strip-path-prefix="/tmp/cov-scan/home" > sast_coverity_buildless_check_all_findings.json
```

Comment:
This is difficult to read.  I would write it like this for better readability:
```sh
csgrep --mode=json --prepend-path-prefix=$SOURCE_CODE_DIR/ $COVERITY_RESULTS_FILE \
    | csgrep --mode=json $SEVERITY_THRESHOLD --remove-duplicates --embed-context 3 \
    | csgrep --mode=json --strip-path-prefix="$SOURCE_CODE_DIR"/source/ \
    | csgrep --mode=json --strip-path-prefix="/tmp/cov-scan/home" \
    > sast_coverity_buildless_check_all_findings.json
```

Reply:
Also `$SEVERITY_THRESHOLD` should be handled by the very first `csgrep` command because it will drop the biggest portion of the results usually.  While the pipe makes it look like the commands are processed in parallel, the JSON format makes the processing effectively serial.  On scans with many findings, the JSON reader may even run out of memory.

Reply:
Modified to be handled by the first csgrep command